app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -494,15 +528,26 @@ object ComponentValidator {
     messages: Messages,
     sse: SmartStringEvaluator
   ) =
-    value match {
-      case tooLong if tooLong.length > maxChars =>
+    (value.filterNot(c => skipValidationsCharacterHexStrings.contains(c.toInt.toHexString.toLowerCase)), messageKey) match {
+      case (tooLong, _) if tooLong.length > maxChars =>
         val vars: List[String] = maxChars.toString :: Nil
-        validationFailure(fieldValue, "generic.error.maxLength", Some(vars))
-      case tooShort if tooShort.length < minChars =>
+        validationFailure(fieldValue, genericErrorMaxLength, Some(vars))
+      case (tooShort, _) if tooShort.length < minChars =>
         val vars: List[String] = minChars.toString :: Nil
-        validationFailure(fieldValue, "generic.error.minLength", Some(vars))
-      case regex() => validationSuccess
-      case _       => validationFailure(fieldValue, messageKey, None)
+        validationFailure(fieldValue, genericErrorMinLength, Some(vars))
+      case (regex(), _) => validationSuccess
+      case (s, `genericLongTextErrorPattern`) =>
```

Comment:
We should let `sharedTextComponentValidator` untouched. Particularly  this case is indication this is not right approach.

 If there is something special to do for `genericLongTextErrorPattern` than let's do it there only ie. in `textValidationWithConstraints`. 

`Validated` is powerful enough to chain multiple validations...  look for `andThen` method.